### PR TITLE
Unfork RCTLogBox

### PR DESCRIPTION
**Pull request by christophpurrer**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

These files have some 'unnecessary' changes which make it hard(er) to merge in recent changes as: https://github.com/facebook/react-native/commit/37e5fa3a6ce 

Hence un-forking them were possible

## Changelog

[macOS][Fixed] Unfork RCTLogBox

## Test Plan

CircleCi for macOS and iOS

Show some beautiful logboxes ...

https://user-images.githubusercontent.com/484044/201546620-6d3bf508-391b-4393-9a40-17322519dc2d.mov

### CODE REVIEW

File: React/CoreModules/RCTLogBoxView.h [hunk 1]
```diff
@@ -11,27 +11,25 @@
 
 #if !TARGET_OS_OSX // TODO(macOS GH#774)
 
-@interface RCTLogBoxWindow : UIWindow // TODO(macOS GH#774) Renamed from _view to _window
+@interface RCTLogBoxView : UIWindow
```
Comment (Saadnajmi) [inline React/CoreModules/RCTLogBoxView.h:14]: Given that in core this is a UIWindow, would it make more sense to upstream our rename instead? This topic came up in previous merges and I wasn't sure if it made sense to.

Comment (christophpurrer) [reply React/CoreModules/RCTLogBoxView.h:14]: I would just KISS ;-) and optimize for 'easy merge'.

There are 5 more commits we have to merge and all cause more pain due to renaming `RCTLogBoxView` to `RCTLogBoxWindow` > https://github.com/facebook/react-native/commits/main/React/CoreModules/RCTLogBoxView.mm

Now:
- Each merge requires to remap `_view` to `_windows` AND
- Factor in the `#ifdef` macOS `#else` ... `#endif` logic

If we would rename it, we also have to rename the file (and the imports) and we would only do it in 0.71 - not solving the immediate problem here.

The most painful resolution would be:
- rename back here
- merge till 0.71
- (if we want, rename back in core in 0.71)

**Comment by Saadnajmi** [comment]
@amgleitman any opinions? I remember you ran into this same diff during one of your last merges.

File: React/CoreModules/RCTLogBoxView.h (hunk 1 above)
Comment (Saadnajmi) [reply React/CoreModules/RCTLogBoxView.h:14]: Fair enough. I logged a task internally for me to followup on this.

**Review by Saadnajmi — APPROVED** [review]

File: React/CoreModules/RCTLogBoxView.h (hunk 1 above)
Comment (christophpurrer) [reply React/CoreModules/RCTLogBoxView.h:14]: You can still rename it in react-native@main branch.
I just would not do it, before all current changes are merged

Comment (Saadnajmi) [reply React/CoreModules/RCTLogBoxView.h:14]: Yep! If anything I'll do it after the 0.71 merge completes, when theoretically we are closer to RN core and it's easier to cherry pick changes across.

Comment (christophpurrer) [reply React/CoreModules/RCTLogBoxView.h:14]: But then RN core might already be at 0.72 or 0.73 ... a race which never can be won ... ;-)

Comment (christophpurrer) [reply React/CoreModules/RCTLogBoxView.h:14]: So better create a PR now ;-)

Comment (Saadnajmi) [reply React/CoreModules/RCTLogBoxView.h:14]: https://github.com/facebook/react-native/pull/35358

Comment (christophpurrer) [reply React/CoreModules/RCTLogBoxView.h:14]: Now or never ;-)

**Pull request merged**